addons/website/static/src/js/editor/snippets.options.js

```diff
@@ -698,10 +698,10 @@ options.registry.gallery = options.Class.extend({
      */
     mode: function (previewMode, value, $li) {
         this.$target.css('height', '');
-        this[value]();
         this.$target
             .removeClass('o_nomode o_masonry o_grid o_slideshow')
             .addClass('o_' + value);
+        this[value]();
```

Comment:
Review the commit message. Don't understand the "and we needed to do it" part :wink: